### DISCUSSION TITLE
Dependency Graph not showing any dependents when there should be

### DISCUSSION THREAD

**Discussion by welkeyever** [community/community#34687] [New to GitHub] [answered]
My repo is only showing dependencies but not dependents. Other repos show in their dependency referencing my repo but it is not showing up as a dependent on mine.

Here is my repo expecting dependents: https://github.com/cloudwego/hertz/network/dependents

Here are some repos referencing my package and show in their dependency graph: 
* https://github.com/hertz-contrib/registry/network/dependencies
* https://github.com/zze326/hertz-nacos-demo/network/dependencies
* https://github.com/stulzq/hexo-deploy-agent/network/dependencies
* ...

**Comment by welkeyever** [comment]
I've searched some similar cases, which ends up a "run a job to redetect" to solve. Seems like an internal operation from Github? 
like these: 
* https://github.com/orgs/community/discussions/23565
* https://github.com/orgs/community/discussions/21637

Am I asking this question in the wrong place? If there is a correct way to raise this kind of question, please let me know~ Thanks!

**Reply by queenofcorgis** [reply]
You came to the right place 😄 I reached out and I have the ability to run the job so I just did so - let me know if this resolves your issue. 

Also really really appreciate you doing some sleuthing and finding these past posts - they were very helpful in leading me to the proper resources.

**Reply by welkeyever** [reply]
I really appreciate your kindly reply~ Up to now, it seems still [not work](https://github.com/cloudwego/hertz/network/dependents) for my repo... If there is anything I need to do, please let me know 🫶

**Reply by queenofcorgis** [reply]
I'm going to try once more, but if this still doesn't work I'm afraid I'm going to have to do some deeper diving 🤿 .

**Reply by arogozhnikov** [reply]
@queenofcorgis can you try the same for [this repo](https://github.com/arogozhnikov/einops) ?
It previously had 4k dependents, and all of a sudden in turned into zero.

**Reply by queenofcorgis** [reply]
@arogozhnikov just queued up the job - please give it a few hours and let me know!

**Reply by arogozhnikov** [reply]
@queenofcorgis  unfortunately still zero...

**Reply by arvkevi** [reply]
Could this be related to removing `setup.py`, which happened recently in [this commit](https://github.com/arogozhnikov/einops/commit/c7a1727510be7f8e094c723fc2e654e4a90957a0)? I've been following this thread because when I switched to primarily using `pyproject.toml` for [kneed](https://github.com/arvkevi/kneed), I noticed shortly after that the dependents went to 0. I've thought about adding the functionality back to `setup.py`, but I'm hesitant because I'm unsure how setuptools, etc. will handle having both pyproject.toml and setup.py -- guess I could test it and read some more documentation.

[Today's blog post](https://github.blog/changelog/2022-10-24-dependabot-updates-support-for-the-python-pep-621-standard/) mentions PEP621 support, if dependabot is used under the hood for dependents, maybe this will fix it?

**Reply by arogozhnikov** [reply]
@arvkevi yes, I think timeline somewhat aligns with removing (deprecated) `setup.py`, that's a very nice catch. Despite post, I still see zero

**Reply by arogozhnikov** [reply]
@queenofcorgis can you check with the team if `setup.py` could be an issue? It's common to see `setup.py` in repos these days, but really... it was deprecated years ago in favor of `pyproject.toml`

**Reply by welkeyever** [reply]
Hi @queenofcorgis ~ Is there anything to update? Seems like it is still not working for [my repo](https://github.com/cloudwego/hertz/network/dependents) 😥

**Reply by arogozhnikov** [reply]
@arvkevi I can absolutely confirm now this is `setup.py`

**Reply by arvkevi** [reply]
Interesting @arogozhnikov are you able to share your evidence with the GH dev team? If you do, and it's public, can you please post a link to the post? I don't really have plans to go back to `setup.py`, rather I want to drop it...

**Comment by li-jin-gou** [comment]
@queenofcorgis  Is there any progress on https://github.com/cloudwego/hertz/network/dependents?

**Comment by li-jin-gou** [comment]
Hello dear @queenofcorgis  Is there any progress on https://github.com/cloudwego/hertz/network/dependents?

I've heard that it seems to work just fine with a bit of backend manipulation, and this has been a problem for months now. Can help if you have time or I'll reintroduce an issue. best wishes.

**Comment by Tishka17** [comment]
I also see no dependents in myprojects.
https://github.com/Tishka17/aiogram_dialog/
https://github.com/reagento/dataclass-factory
https://github.com/reagento/dataclass-rest

Even though I can see, that dataclass-rest depends on dataclass-factory

**Comment by queenofcorgis** [comment]
Hi there ! This has been opened as an internal Issue for our Dependency Graph team - it is on their radar, but there is no timeline. Will share updates when I have them. If there is anything you'd like to include in the Issue, please let me know. Otherwise, I appreciate your patience.

**Reply by li-jin-gou** [reply]
> Hi there ! This has been opened as an internal Issue for our Dependency Graph team - it is on their radar, but there is no timeline. Will share updates when I have them. If there is anything you'd like to include in the Issue, please let me know. Otherwise, I appreciate your patience.

thanks ❤️

**Reply by welkeyever** [reply]
Really appreciate it! Looking forward to the follow-up update🙏

**Reply by aminalaee** [reply]
@queenofcorgis Has there been any progress with this?

**Reply by queenofcorgis** [reply]
Thanks for the reminder - I just followed back up with the team - when I have updates I will share them.

**Comment by CalMacCQ** [comment]
Hi there, 

Can I ask how github determines the repository associated with a particular dependency? I've encountered an issue which seems similar to those described above.

The source code of the pytket python library is contained in this repository -> https://github.com/CQCL/tket
However the list of repositories which depend on this project is empty which is unexpected -> https://github.com/CQCL/tket/network/dependents.

One of the several repos I would expect to depend on this one is this `pytket-qiskit` repo. https://github.com/CQCL/pytket-qiskit/network/dependents. Here the dependency graph seems to work for both dependencies and dependents but it lists CQCL/pytket as a dependency instead of CQCL/tket as I would expect. Perhaps confusingly CQCL/pytket contains only documentation and notebook examples and the source code for the pytket package is in CQCL/tket.

Maybe this is caused by the less than ideal naming of the different repositories or the fact that [setup.py](https://github.com/CQCL/tket/blob/develop/pytket/setup.py) is used to list the dependencies in the tket repository instead of `requirements.txt` or `pipfile.lock`.

Apologies for the long comment, any help with this would be much appreciated :)

**Reply by ducha-aiki** [reply]
Same problems for us -- https://github.com/kornia/kornia. However, we do have `setup.py`...

**Reply by WenjieDu** [reply]
Well, I have the same question with @CalMacCQ. I tried but didn't find the answer from GitHub official docs. 
Could you @queenofcorgis please kindly make an explanation? Many thanks. 😃

**Comment by alexdlaird** [comment]
I have the same issue on two of my projects, one Python ([pyngrok](https://github.com/alexdlaird/pyngrok)), one Java ([java-ngrok](https://github.com/alexdlaird/java-ngrok)). Neither show "Depends" or "Used By". My Python package is using setup.py and not pyproject.toml, but I know of other packages that haven't migrated that still show a Used By ([example](https://github.com/encode/django-rest-framework)). Any update on this? Is GitHub just not supporting Dependabot anymore? I tried opening a support ticket, no response.

**Comment by queenofcorgis** [comment]
Hi there - the team that worked on this reported that this should be fixed - please let me know if this is the case for y'all.

**Reply by alexdlaird** [reply]
Not resolved in either of my projects, unless there is some time that needs to pass for it to re-find dependents?

https://github.com/alexdlaird/pyngrok
https://github.com/alexdlaird/java-ngrok

**Reply by aminalaee** [reply]
Also not working for my project:

https://github.com/aminalaee/sqladmin
https://github.com/aminalaee/sqladmin-demo

**Reply by aminalaee** [reply]
@queenofcorgis thanks for taking a look into this.

**Reply by welkeyever** [reply]
Thanks a lot, it works for my project🙏

**Comment by CalMacCQ** [comment]
This hasn't updated for my project yet either. Thanks for looking into this.

https://github.com/CQCL/tket/network/dependents

**Reply by bgu6667** [reply]
[]()

**Comment by StevenClontz** [comment]
Our repo at https://github.com/PreTeXtBook/pretext-cli/network/dependents shows no dependents for the `pretext` pip package, but the stale/archived repo (whose owner kindly allowed us to take over the name) does show our dependents: https://github.com/moreati/b-prefix-all-the-doctests/network/dependents

**Comment by kayhantolga** [comment]
Unfortunately same here https://github.com/betalgo/openai

**Comment by StevenClontz** [comment]
Is there a way to mark this as unanswered as it seems to not be working for multiple users still?

**Reply by queenofcorgis** [reply]
I'll let the team investigating this know and I've unmarked as answer. Thanks!

**Reply by aminalaee** [reply]
Thank you @queenofcorgis

**Comment by queenofcorgis** [comment, marked as answer]
Hi everyone - thanks for your patience and for coming to the community after some more digging I've gotten more clarity. These fixes are actually more successful when done on a case-by-case basis so if you are seeing this issue, you will have to open up a [support ticket](https://support.github.com/contact). Be sure to include which repository(ies) is/are affected in your ticket. Thanks again!

**Reply by StevenClontz** [reply]
Confirmed - my repo is now fixed after opening a support ticket. (I think I chose the topic "restore deleted data" but I'm not sure.)

**Reply by Jack-Edwards** [reply]
Which category on the support page are we supposed to choose?  Nothing seems close to appropriate for this.

**Reply by andrzejkupczyk** [reply]
@Jack-Edwards , the "Other Support" category appears to be a promising option
